src/pages/registration/references_tab/components/SearchContainerField.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { Field, FieldProps, getIn, useFormikContext } from 'formik';
+import { Typography } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import Truncate from 'react-truncate';
+import { Registration } from '../../../../types/registration.types';
+import { autocompleteTranslationProps } from '../../../../themes/mainTheme';
+import useSearchRegistrations from '../../../../utils/hooks/useSearchRegistrations';
+import { AutocompleteTextField } from '../../../../components/AutocompleteTextField';
+import EmphasizeSubstring from '../../../../components/EmphasizeSubstring';
+import { StyledFlexColumn } from '../../../../components/styled/Wrappers';
+import { displayDate } from '../../../../utils/date-helpers';
+import useDebounce from '../../../../utils/hooks/useDebounce';
+import { API_URL } from '../../../../utils/constants';
+import { getRegistrationPath } from '../../../../utils/urlPaths';
+import { createSearchQuery } from '../../../../utils/searchHelpers';
+import { ReferenceFieldNames, RegistrationSubtype } from '../../../../types/publicationFieldNames';
+
+interface SearchContainerFieldProps {
+  fieldName: string;
+  searchSubtypes: RegistrationSubtype[];
+  label: string;
+  placeholder: string;
+}
+
+const SearchContainerField = (props: SearchContainerFieldProps) => {
+  const { values, setFieldValue, setFieldTouched } = useFormikContext<Registration>();
+  const [searchTerm, setSearchTerm] = useState('');
+  const debouncedSearchTerm = useDebounce(searchTerm);
+
+  const optionsSearchQuery = createSearchQuery({
+    searchTerm: debouncedSearchTerm,
+    properties: [{ key: ReferenceFieldNames.SUB_TYPE, value: props.searchSubtypes }],
```

Comment:
dette er basically databaseoppslag ikke sant? synes det er litt forvirrende å kalle det `key`. er det mulig å endre til f.eks `filter` eller `category` eller noe lignende? evt at det kommer frem på en annen måte at dette er en query? kanskje isteden for `properties` så kan det hete `filter`?

Reply:
dette er søket som går mot elasticsearch, så det blir på en måte et DBoppslag ja.
Vi søker på properties til Registration-objekter, derfor kom det navnet.
Eks: Vi vil ha Registreringer som har property `identifier: xxx`
Det er kanskje bedre å kalle `key` for `fieldName`? (`key` er i praksis det samme som brukes for `name`-prop i Formik Field)

Reply:
Ja, jeg synes fieldName er bedre!